packages/frontend/src/components/MetricQueryData/DrillDownModal.tsx

```diff
@@ -203,43 +201,40 @@ const DrillDownModal: FC = () => {
     }, [closeDrillDownModal]);
 
     return (
-        <Dialog
-            isOpen={isDrillDownModalOpen}
+        <Modal
+            opened={isDrillDownModalOpen}
             onClose={closeDrillDownModal}
-            lazy
-            title={`Drill into "${value}"`}
+            title={<Title order={4}>{`Drill into "${value}"`}</Title>}
         >
-            <form>
-                <div className={Classes.DIALOG_BODY}>
-                    <FormGroup
-                        label="Pick a dimension to segment your metric by"
-                        labelFor="chart-name"
+            <Stack>
+                <FieldSelect
+                    withinPortal
+                    disabled={dimensionsAvailable.length === 0}
+                    item={selectedDimension}
+                    items={dimensionsAvailable}
+                    onChange={setSelectedDimension}
+                />
+                <Group position="right">
+                    <Button variant="outline" onClick={onClose}>
+                        Cancel
+                    </Button>
+                    <Button
+                        component="a"
+                        target="_blank"
+                        href={url}
+                        leftIcon={<MantineIcon icon={IconExternalLink} />}
+                        disabled={!selectedDimension}
+                        // TODO: fix this timeout
+                        onMouseUpCapture={() =>
```

Comment:
simple onClick does not work? 🤔

Reply:
nope!

Reply:
There's likely a bug with Mantine's polymorphic buttons where the onclick event triggers before the navigation, resulting in an empty URL or something similar. It's not worth delving further into it. I think a 500ms timeout is fine, as the menu closing happens in the background.